### PR TITLE
feat(applying): add missing validation checks, prot params feature and documentation

### PR DESCRIPTION
**Pull request by MaicoLeberle**

### CODE REVIEW

File: pallas-applying/Cargo.toml [hunk 1]
```diff
@@ -13,8 +13,14 @@ authors = ["Maico Leberle <maico.leberle@gmail.com>"]
 doctest = false
 
 [dependencies]
+cryptoxide = "0.1"
+cbor_event = "^2.1.1"
```
Comment (scarmuega) [inline pallas-applying/Cargo.toml:17]: we already have a crate for dealing with cbor serialization, it's called `minicbor`. You can access it via `pallas::codec::minicbor`. Any particular reason we need `cbor_event`?

File: pallas-applying/src/byron.rs [hunk 2]
```diff
@@ -38,3 +58,183 @@ fn check_ins_in_utxos(tx: &Tx, utxos: &UTxOs) -> ValidationResult {
...
+        let (pub_key, sign): (&PubKey, &TaggedSignature) = find_witness(tx_out, &witnesses)?;
+        let signature: &Vec<u8> = sign.get_raw_signature();
+        let mut se = Serializer::new_vec();
+        serialize_tag(&mut se, sign)?;
+        se.serialize(&prot_magic)
+            .map_err(|_| ValidationError::UnableToProcessWitnesses)?;
+        se.serialize(&tx_hash.as_ref())
+            .map_err(|_| ValidationError::UnableToProcessWitnesses)?;
+        let data_to_verify: Vec<u8> = se.finalize();
+        if !ed25519::verify(&data_to_verify, &pub_key.as_slice()[0..32], signature) {
```
Comment (scarmuega) [inline pallas-applying/src/byron.rs:139]: we already have a function in `pallas::crypto::keys` to verify ed25519. We should use that one.

File: pallas-applying/Cargo.toml [hunk 3]
```diff
@@ -13,8 +13,14 @@ authors = ["Maico Leberle <maico.leberle@gmail.com>"]
 doctest = false
 
 [dependencies]
+cryptoxide = "0.1"
```
Comment (scarmuega) [inline pallas-applying/Cargo.toml:16]: crypto algos should be made available through `pallas::crypto`, which already depends on `cryptoxide`.

File: pallas-applying/tests/byron.rs [hunk 4]
```diff
@@ -11,77 +10,99 @@ use pallas_codec::{
...
-        add_tx_out(&mut tx_outs, &tx_out);
+    const MAINNET_TX_HEX_CBOR: &str =
+        "82839f8200d8185824825820da832fb5ef57df5b91817e9a7448d26e92552afb34f8ee5adb491b24bbe990d50e\
+        ff9f8282d818584283581cdac5d9464c2140aeb0e3b6d69f0657e61f51e0c259fe19681ed268e8a101581e581c2\
+        b5a44277e3543c08eae5d9d9d1146f43ba009fea6e285334f2549be001ae69c4d201b0000000172a84e408282d8\
+        18584283581c2b8e5e0cb6495ec275872d1340b0581613b04a49a3c6f2f760ecaf95a101581e581cca3e553c9c6\
+        3c5b66689e943ce7dad7d560ae84d7c2eaf21611c024c001ad27c159a1b00000003355d95efffa0818200d81858\
+        85825840888cdf85991d85f2023423ba4c80d41570ebf1fc878c9f5731df1d20c64aecf3e8aa2bbafc9beba8ef3\
+        3acb4d7e199b445229085718fba83b7f86ab6a3bcf782584063e34cf5fa6d8c0288630437fa5e151d93907e826e\
+        66ba273145e3ee712930b6f446ff81cb91d7f0cb4ceccd0466ba9ab14448d7eab9fc480a122324bd80170e";
```
Comment (scarmuega) [inline pallas-applying/tests/byron.rs:30]: Not required for now, but we should load test data from the `test_data` folder where we already hold test vectors for different crates.

File: pallas-applying/Cargo.toml (hunk 1 above)
Comment (MaicoLeberle) [reply pallas-applying/Cargo.toml:17]: I used `cbor_event` trying to mimick what's done in [rust-byron-cardano](https://github.com/input-output-hk/rust-byron-cardano/blob/5f0c8d34424e0ccb199b8c6b70fd410d4bd6af41/cardano/src/tx.rs#L148), specifically the part concerned with preparing the bytes array on which the witness signatures are verified. I can try using minicbor instead, and see if I can still check witnesses

File: pallas-applying/Cargo.toml (hunk 3 above)
Comment (MaicoLeberle) [reply pallas-applying/Cargo.toml]: Just added a [commit](https://github.com/txpipe/pallas/commit/d74b1484e91b8c417a59442cae9b3d96caafbd81) correcting this.

File: pallas-applying/tests/byron.rs (hunk 4 above)
Comment (MaicoLeberle) [reply pallas-applying/tests/byron.rs]: In [this commit](https://github.com/txpipe/pallas/commit/aeb1a1a268a901cae201570637e831dec01a1b42) I moved the CBOR to test_data

File: pallas-applying/Cargo.toml (hunk 1 above)
Comment (MaicoLeberle) [reply pallas-applying/Cargo.toml]: I was just trying to copy the implementation in [rust-byron-cardano](https://github.com/input-output-hk/rust-byron-cardano/blob/5f0c8d34424e0ccb199b8c6b70fd410d4bd6af41/cardano/src/tx.rs#L148). But nevermind, I just added a [commit](cf3067f6eb3e362e24d2f19e31db204cda92564e) changing it `minicbor` instead.

**Review by scarmuega — APPROVED** [review]

**Pull request merged**